File: src/main/java/flustix/fluxifyed/console/ConsoleCommands.java
package flustix.fluxifyed.console;

import flustix.fluxifyed.console.commands.HelpConsoleCommand;
import flustix.fluxifyed.console.commands.StopConsoleCommand;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.io.BufferedReader;
import java.util.Arrays;
import java.util.HashMap;

public class ConsoleCommands {
    static HashMap<String, ConsoleCommand> commands = new HashMap<>();
    public static Logger LOGGER = LoggerFactory.getLogger("Flux Console");

    public static void start() {
        registerCommand(new StopConsoleCommand());
        registerCommand(new HelpConsoleCommand());

        while (true) {
            listen();
        }
    }

    static void registerCommand(ConsoleCommand command) {
        commands.put(command.name, command);
        LOGGER.debug("Registered console command " + command.name);
    }

    public static void listen() {
        try {
            BufferedReader reader = new BufferedReader(new java.io.InputStreamReader(System.in));
            String line = reader.readLine();
            execute(line);
        } catch (Exception ex) {
            LOGGER.error("Error while reading console input: " + ex.getMessage(), ex);
        }
    }

    static void execute(String line) {
        String[] args = line.split(" ");

        if (args.length == 0) {
            return;
        }

        String commandName = args[0];
        String[] commandArgs = Arrays.copyOfRange(args, 1, args.length);

        if (commands.containsKey(commandName)) {
            commands.get(commandName).execute(commandArgs);
        } else {
            LOGGER.error("Unknown command " + commandName);
        }
    }

    public static HashMap<String, ConsoleCommand> getCommands() {
        return commands;
    }
}

File: Fluxifyed/CC/ConsoleCommands.cs
using Fluxifyed.CC.Commands;
using Fluxifyed.Logging;

namespace Fluxifyed.CC; 

public static class ConsoleCommands {
    public static readonly List<IConsoleCommand> Commands = new();
    private static bool _run = true;

    public static void Init() {
        Register(new HelpConsoleCommand());
        Register(new GuildsConsoleCommand());
        Register(new StopConsoleCommand());
        Register(new SayConsoleCommand());
        Register(new ReplyConsoleCommand());
        Register(new ModulesConsoleCommand());
        
        Commands.Sort((a, b) => string.Compare(a.Name, b.Name, StringComparison.Ordinal));
        
        var thread = new Thread(Listen) { IsBackground = true };
        thread.Start();
    }
    
    private static void Register(IConsoleCommand command) {
        // Logger.Log($"Registered console command '{command.Name}'");
        Commands.Add(command);
    }
    
    public static void Stop() {
        _run = false;
    }

    private static void Listen() {
        while (_run) {
            var input = Console.ReadLine();

            if (input == null) continue;
            var split = input.Split(" ");
            var args = split.Skip(1).ToArray();
                
            var commandName = split[0];
            var command = Commands.FirstOrDefault(c => c.Name == commandName);

            try {
                command?.Execute(args);
                    
                if (command == null) {
                    Console.ForegroundColor = ConsoleColor.Red;
                    Console.Write($"Command '{commandName}' not found!\n");
                    Logger.EmptyLine();
                }
            }
            catch (Exception e) {
                Console.ForegroundColor = ConsoleColor.Red;
                Console.Write($"Error: {e.Message}\n");
                Logger.EmptyLine();
            }
        }
    }
}